src/compiler/emitter.ts

```diff
@@ -4046,7 +4006,7 @@ module ts {
                 var preambleEmitted = writer.getTextPos() !== outPos;
 
                 if (!preambleEmitted && nodeEndIsOnSameLineAsNodeStart(body, body)) {

```

Comment:
Actually I think this is precisely what you are handling that the previous code did not.